isobus/include/isobus/isobus/isobus_task_controller_server.hpp

```diff
@@ -0,0 +1,534 @@
+			Reserved = 0x0B, ///< Reserved.
+			Reserved2 = 0x0C, ///< Reserved.
+			Acknowledge = 0x0D, ///< Message is a Process Data Acknowledge (PDACK).
+			Status = 0x0E, ///< Message is a Task Controller Status message
+			ClientTask = 0x0F ///< Sent by the client as a status message every 2s
+		};
+
+		/// @brief Enumerates the different options that can be reported by the server.
+		/// Each option is a bit in a bitfield, with 1 meaning the option is supported and 0 meaning it is not.
+		/// For example, if the server supports documentation and peer control assignment, but not the other options,
+		/// the bitfield would be 0b00001001.
+		enum class ServerOptions : std::uint8_t
+		{
+			SupportsDocumentation = 0x01,
+			SupportsTCGEOWithoutPositionBasedControl = 0x02,
+			SupportsTCGEOWithPositionBasedControl = 0x04,
+			SupportsPeerControlAssignment = 0x08,
+			SupportsImplementSectionControl = 0x10,
+			ReservedOption1 = 0x20,
+			ReservedOption2 = 0x40,
+			ReservedOption3 = 0x80
+		};
+
+		/// @brief Enumerates all PDACK error codes that can be sent to or from the client.
+		enum class ProcessDataAcknowledgeErrorCodes : std::uint8_t
+		{
+			ProcessDataCommandNotSupported = 0x01,
+			InvalidElementNumber = 0x02,
+			DDINotSupportedByElement = 0x04,
+			TriggerMethodNotSupported = 0x08,
+			ProcessDataNotSettable = 0x10,
+			InvalidOrUnsupportedIntervalOrThreshold = 0x20,
+			ProcessDataValueDoesNotConformToDDIDefinition = 0x40,
+			ProcessDataValueIsOutOfOperationalRangeOfThisDevice = 0x80
+		};
+
+		/// @brief Constructor for a TC server.
+		/// @param internalControlFunction The control function to use to communicate with the clients.
+		/// @param numberBoomsSupported The number of booms to report as supported by the TC.
+		/// @param numberSectionsSupported The number of sections to report as supported by the TC.
+		/// @param numberChannelsSupportedForPositionBasedControl The number of channels to report as supported by the TC.
+		/// @param optionsBitfield The options bitfield to report as supported by the TC. See ServerOptions enum for the bit definitions.
+		TaskControllerServer(std::shared_ptr<InternalControlFunction> internalControlFunction,
+		                     std::uint8_t numberBoomsSupported,
+		                     std::uint8_t numberSectionsSupported,
+		                     std::uint8_t numberChannelsSupportedForPositionBasedControl,
+		                     std::uint8_t optionsBitfield);
+
+		/// @brief Destructor for a TC server.
+		virtual ~TaskControllerServer();
+
+		/// @brief Deleted copy constructor
+		TaskControllerServer(TaskControllerServer &) = delete;
+
+		/// @brief Deleted assignment operator
+		TaskControllerServer &operator=(const TaskControllerServer &) = delete;
+
+		/// @brief Shuts down the TC server, unregisters PGN callbacks
+		void terminate();
```

Comment:
Nit: would be more clear if we move this function down to the (de)init functions